exporter/azuremonitorexporter/factory.go

```diff
@@ -44,7 +44,8 @@ func NewFactory() component.ExporterFactory {
 		typeStr,
 		createDefaultConfig,
 		component.WithTracesExporter(f.createTracesExporter, stability),
-		component.WithLogsExporter(f.createLogsExporter, stability))
+		component.WithLogsExporter(f.createLogsExporter, stability),
+		component.WithMetricsExporter(f.createMetricsExporter, stability))
```

Comment:
I think we may also want to make this feature opt-in explicitly. Cost can be a real concern for Azure Monitor customers. Additionally, people may NOT want to send any metrics to Azure Monitor and instead rely on some other backend/exporter like the Prometheus exporter to expose metrics.

Reply:
I wouldn't do that. If people don't want to send metrics to Azure Monitor, they don't define that in their pipeline. Opting in metrics explicitly will introduce extra configuration complexity and weirdness (e.g. why should I config metrics exporter in a different way than traces and logs?). That could have some logic with the traces exporter as actually traces is the most expensive data stream. But still, I wouldn't make make people treat exporters differently.

Reply:
I'm still concerned that we aren't giving consumers enough freedom to enable/disable certain features. Here is an example of what my team might do with this exporter. We currently have 2 collector deployments, 1 that does trace grouping and another that does tail sampling. We also use the Prometheus exporter to publish metrics about ALL the requests, before the traces are sampled. We use this Azure Monitor exporter after tail sampling is complete. If we were to start pushing our metrics to Azure Monitor, we would have to turn off traces, logs, and just enable the metrics. So I would argue that ALL the exporter features should have knobs. Traces came first, and doesn't have one, but it should and so should metrics.

Reply:
@pcwiese I think you turn on metrics by adding this receiver to the metrics pipelines of your collector. No metrics pipelines, no cost around metrics. Am I missing something?

Reply:
@atoulme self head slap! You are right. I don't know what I was thinking. Ignore this :)